webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -204,85 +250,100 @@ trait CachedFindRuleNodeStatusReports extends ReportingService with CachedReposi
+         *    ok and will be until a new report comes)
          * 2/ cache exists but expiration date expired,
          * 3/ cache does note exists.
+         *
          * For simplicity (and keeping computation logic elsewhere than in a cache that already has its cache logic
          * to manage), we will group 2 and 3, but it is well noted that it seems that a RuleNodeStatusReports that
          * expired could be computed without any more logic than "every thing is missing".
          *
          * The definition of expired date is the following:
          *  - Node is Pending -> expirationDateTime is the expiration time
          *  - There is a LastRunAvailable -> expirationDateTime is the lastRunExpiration
-         *  - Other cases: no expiration
+         *  - Other cases: no expiration, ie a "missing report" can not expire (and that's what we want)
          *
          */
-        (expired, upToDate) =  nodeIds.partition { id =>  //two group: expired id, and up to date info
-                                 cache.get(id) match {
-                                   case None         => true
-                                   case Some(status) => status.runInfo match { // We have a status on the run
-                                     case t : ExpiringStatus => t.expirationDateTime.isBefore(now)
-                                     case UnexpectedVersion(_, _, lastRunExpiration, _, _)
-                                                             => lastRunExpiration.isBefore(now)
-                                     case UnexpectedNoVersion(_, _, lastRunExpiration, _, _)
-                                                             => lastRunExpiration.isBefore(now)
-                                     case _                  => false
-                                   }
-                                 }
-                               }
-        newStatus           <- defaultFindRuleNodeStatusReports.findRuleNodeStatusReports(expired, Set())
+         upToDate     =  inCache.filter { case (_, report) =>
+                           val expired = report.runInfo match {
+                             case t : ExpiringStatus => t.expirationDateTime.isBefore(now)
+                             case UnexpectedVersion(_, _, lastRunExpiration, _, _)
+                                                     => lastRunExpiration.isBefore(now)
+                             case UnexpectedNoVersion(_, _, lastRunExpiration, _, _)
+                                                     => lastRunExpiration.isBefore(now)
+                             case _                  => false
+                           }
+                           !expired
+                         }
+        // starting with nodeIds, is all accepted node passed in parameter,
+        // we don't miss node ids not yet in cache
+        requireUpdate =  nodeIds -- upToDate.keySet
+        _             <- invalidate(requireUpdate).unit.toBox
```

Comment:
this will never return compliance for node with unexpected reports (like nodes with broken policies)